app/src/main/java/org/mozilla/fenix/compose/cfr/CFRPopup.kt

```diff
@@ -0,0 +1,610 @@
+import org.mozilla.fenix.R
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.gecko.GeckoScreenOrientation
+import java.lang.ref.WeakReference
+import kotlin.math.absoluteValue
+import kotlin.math.roundToInt
+
+/**
+ * Fixed width for all CFRs.
+ */
+private const val POPUP_WIDTH_DP = 335
+
+/**
+ * Fixed horizontal padding.
+ * Allows the close button to extend with 10dp more to the end and intercept touches to a bit outside of the popup
+ * to ensure it respects a11y recomendations of 48dp size while also offer a bit more space to the text.
+ */
+private const val HORIZONTAL_PADDING = 10
+
+/**
+ * Value class allowing to easily reason about what an `Int` represents.
+ * This is compiled to the underlying `Int` type so incurs no performance penalty.
+ */
+@JvmInline
+internal value class Pixels(val value: Int)
+
+/**
+ * Simple wrapper over the absolute x-coordinates of the popup. Includes any paddings.
+ */
+internal data class PopupHorizontalBounds(
+    val startCoord: Pixels,
+    val endCoord: Pixels
+)
+
+/**
+ * Possible direction for the arrow indicator of a CFR popup.
+ * The direction is expressed in relation with the popup body containing the text.
+ */
+enum class IndicatorDirection {
+    UP,
+    DOWN
+}
+
+/**
+ * Properties used to customize the behavior of a [CFRPopup].
+ *
+ * @property indicatorDirection The direction the indicator arrow is pointing to.
+ * @property dismissOnBackPress Whether the popup can be dismissed by pressing the back button.
+ * If true, pressing the back button will also call onDismiss().
+ * @property dismissOnClickOutside Whether the popup can be dismissed by clicking outside the
+ * popup's bounds. If true, clicking outside the popup will call onDismiss().
+ * @property overlapAnchor How the popup's indicator will be shown in relation to the anchor:
+ *   - true - indicator will be shown exactly in the middle horizontally and vertically
+ *   - false - indicator will be shown horizontally in the middle of the anchor but immediately below or above it
+ * @property indicatorArrowStartOffset Maximum distance between the popup start and the indicator arrow.
+ * If there isn't enough space this could automatically be overridden up to 0.
+ * @property indicatorArrowHeight How tall the indicator arrow should be.
+ * This will also affect how wide the base of the indicator arrow will be.
+ * @property cornerRadius Corner radius for the popup body.
+ * @property elevation The z-coordinate at which to place the popup.
+ * This controls the size of the shadow below the popup.
+ */
+data class CFRPopupProperties(
+    val indicatorDirection: IndicatorDirection = IndicatorDirection.UP,
+    val dismissOnBackPress: Boolean = true,
+    val dismissOnClickOutside: Boolean = true,
+    val overlapAnchor: Boolean = false,
+    val indicatorArrowStartOffset: Dp = 30.dp,
+    val indicatorArrowHeight: Dp = 10.dp,
+    val cornerRadius: Dp = 10.dp,
+    val elevation: Dp = 10.dp,
```

Comment:
where are the default specs for this obtained from?

Reply:
Thank you for raising this.
I just eyeballed the values to get everything working but did not came back to polish.
Checked now in previous CFRs  ([onboarding_jump_back_in_cfr](https://github.com/mozilla-mobile/fenix/blob/263e4d819be3b67c037917237f55050f3c09b8c9/app/src/main/res/layout/onboarding_jump_back_in_cfr.xml) and [onboarding_inactive_tabs_cfr](https://github.com/mozilla-mobile/fenix/blob/263e4d819be3b67c037917237f55050f3c09b8c9/app/src/main/res/layout/onboarding_inactive_tabs_cfr.xml)) and saw that they use [`15` as the "indicator offset"](https://github.com/mozilla-mobile/fenix/blob/263e4d819be3b67c037917237f55050f3c09b8c9/app/src/main/res/layout/onboarding_jump_back_in_cfr.xml#L55) and [`12` as the cornerRadius](https://github.com/mozilla-mobile/fenix/blob/51715c27b9f84a42b65be14ba3a0bf1cb9864e39/app/src/main/res/drawable/onboarding_popup_shape.xml#L6). Will update accordingly.
The other two properties - `indicatorArrowHeight` and `elevation` remain to be validated by UX maybe when I'll have the TCP CFR PR ready which would use this new CFR.